src/test/java/org/broadinstitute/hellbender/tools/recalibration/CycleCovariateUnitTest.java

```diff
@@ -82,4 +82,8 @@ public void testMaxCyclePasses() {
         ReadCovariates readCovariates = new ReadCovariates(read.getReadLength(), 1);
         covariate.recordValues(read, readCovariates);
     }
+
+    public static int expectedCycle(SAMRecord read, final int baseNumber, final boolean indel, final int maxCycle) {
+        return CycleCovariate.cycleFromKey(CycleCovariate.cycleKey(baseNumber, read, indel, maxCycle));
```

Comment:
Here too, you're using a method from the covariate class to compute expected values. But where are the unit tests testing the correctness of `cycleKey()` and `cycleFromKey()` themselves?